lib/releases.js

```diff
@@ -114,27 +116,63 @@ const categorizePullRequests = (pullRequests, categories) => {
   )
 }
 
-const templateVersion = (template, version) => {
-  return template
-    .replace('$MAJOR', major(version))
-    .replace('$MINOR', minor(version))
-    .replace('$PATCH', patch(version))
+const generateChangeLog = (mergedPullRequests, config) => {
+  const [categoriesConfig, orderedCategories] = getCategoriesConfig({ config })
+  const categorizedPullRequests = categorizePullRequests(
+    mergedPullRequests,
+    orderedCategories
+  )
+  return orderedCategories
+    .reduce((acc, category, index) => {
+      if (
+        !categorizedPullRequests[category] ||
+        categorizedPullRequests[category].length === 0
+      )
+        return acc
+
+      if (category !== UNCATEGORIZED) {
+        acc.push(`## ${categoriesConfig[category].title}\n\n`)
+      }
+
+      acc.push(
+        categorizedPullRequests[category]
+          .map(pullRequest =>
+            template(config['change-template'], {
+              $TITLE: pullRequest.title,
+              $NUMBER: pullRequest.number,
+              $AUTHOR: pullRequest.user.login
+            })
+          )
+          .join('\n')
+      )
+
+      if (index + 1 !== orderedCategories.length) acc.push('\n\n')
+
+      return acc
+    }, [])
+    .join('')
 }
 
-const templateNextVersion = (template, nextVersions, versionTemplate) => {
-  return template
-    .replace(
-      '$NEXT_MAJOR_VERSION',
-      templateVersion(versionTemplate, nextVersions.incrementedMajor) || ''
-    )
-    .replace(
-      '$NEXT_MINOR_VERSION',
-      templateVersion(versionTemplate, nextVersions.incrementedMinor) || ''
-    )
-    .replace(
-      '$NEXT_PATCH_VERSION',
-      templateVersion(versionTemplate, nextVersions.incrementedPatch) || ''
-    )
+const templateVersion = (input, version) => {
+  return template(input, {
+    $MAJOR: major(version),
```

Comment:
I think this reads cleaner instead of `.replace('$MAJOR', major(version)`

Reply:
sure, looks good to me!